nntrainer/layers/layer_context.cpp

```diff
@@ -567,35 +568,94 @@ bool RunLayerContext::validate(bool skip_input, bool skip_label) {
 }
 
 #ifdef ENABLE_OPENCL
+
+/**
+ * @brief Global bit mask to check if kernel already initialized.
+ */
+unsigned int RunLayerContext::kernelInitializedMask = 0;
+
+/**
+ * @brief create OpenCl kernel
+ * @param kernel_string implementation string
+ * @param layerKernel LayerKernel
+ * @return true if kernel creation is successful, false otherwise
+ */
 bool RunLayerContext::clCreateKernel(std::string kernel_string,
-                                     std::string kernel_name) {
-  if (kernel_initialized) {
-    ml_logi("Kernel already initialized: %s", kernel_name.c_str());
+                                     LayerKernel layerKernel,
+                                     opencl::Kernel &kernel_) {
+
+  // checking bitmask for already initialized kernel. eg: 010 & 000 -> 0 but 010
+  // & 110 -> 010
+  if (layerKernel == (kernelInitializedMask & layerKernel)) {
+    ml_logi("Kernel already initialized: %s",
+            getKernelName(layerKernel).c_str());
     return true;
   }
 
-  ml_logi("Kernel initializing: %s", kernel_name.c_str());
+  ml_logi("Kernel initializing: %s", getKernelName(layerKernel).c_str());
 
   bool result = false;
 
   do {
     opencl::Program program;
-    result =
-      program.CreateCLProgram(context_inst_.GetContext(),
-                              context_inst_.GetDeviceId(), kernel_string, "");
+
+    // reading binary
+    std::ifstream fs(opencl::Program::DEFAULT_KERNEL_PATH + "/" +
+                       getKernelName(layerKernel) + "_kernel.bin",
+                     std::ios::binary | std::ios::in);
+
+    if (fs) {
```

Comment:
```suggestion
    if (fs.good()) {
```

Is there any case that ```fs``` can be null?

Reply:
I don't think the `fs` object can be null.